oap-server/server-storage-plugin/storage-influxdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/influxdb/InfluxClient.java

```diff
@@ -0,0 +1,194 @@
+import org.apache.skywalking.oap.server.core.analysis.TimeBucket;
+import org.apache.skywalking.oap.server.library.client.Client;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.influxdb.querybuilder.time.TimeInterval;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.influxdb.querybuilder.BuiltQuery.QueryBuilder.ti;
+
+/**
+ *
+ */
+public class InfluxClient implements Client {
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private InfluxStorageConfig config;
+    private InfluxDB influx;
+
+    /**
+     * A constant, the name of time field in Time-series database.
+     */
+    public static final String TIME = "time";
+    /**
+     * A constant, the name of tag.
+     */
+    public static final String TAG_ENTITY_ID = "entity_id";
+
+    private final String database;
+
+    public InfluxClient(InfluxStorageConfig config) {
+        this.config = config;
+        this.database = config.getDatabase();
+    }
+
+    public final String getDatabase() {
+        return database;
+    }
+
+    @Override
+    public void connect() {
+        influx = InfluxDBFactory.connect(config.getUrl(), config.getUser(), config.getPassword(),
+            new OkHttpClient.Builder(), InfluxDB.ResponseFormat.MSGPACK);
+        influx.query(new Query("CREATE DATABASE " + database));
+
+        influx.setDatabase(database);
+        influx.enableBatch();
+    }
+
+    /**
+     * To get a connection of InfluxDB
+     *
+     * @return InfluxDB's connection
+     */
+    public InfluxDB getInflux() {
+        return influx;
+    }
+
+    /**
+     * Request with a {@link Query} to InfluxDB and return a set of {@link QueryResult.Result}s.
+     *
+     * @param query
+     * @return a set of Result.
+     * @throws IOException
+     */
+    public List<QueryResult.Result> query(Query query) throws IOException {
+        if (logger.isDebugEnabled()) {
+            logger.debug("SQL Statement: {}", query.getCommand());
+        }
+
+        try {
+            QueryResult result = getInflux().query(query);
+            if (result.hasError()) {
+                throw new IOException(result.getError());
+            }
+            return result.getResults();
+        } catch (Exception e) {
+            throw new IOException(e.getMessage() + System.lineSeparator() + "SQL Statement: " + query.getCommand(), e);
+        }
+    }
+
+    /**
+     * Request with one statement to InfluxDB and return a set of {@link QueryResult.Series}s.
+     *
+     * @param query
+     * @return a set of Series
+     * @throws IOException
+     */
+    public List<QueryResult.Series> queryForSeries(Query query) throws IOException {
+        return query(query).get(0).getSeries();
```

Comment:
What does this `#get(0)` represent?

Reply:
InfluxDB support user to send a request with multi statements. This is for a single statement in a request(query).
So, `#get(0)` means get the first result of the query.

Reply:
Do we use multiple statements? I think `#get(0)` should be included `#query`(return series). If we have multiple stat case, we should have a different API.

Reply:
Yes.
In `ITraceQueryDAO#queryBasicTraces(...)`, we have 2 statements in a request. The one gets the number of traces, and another recalls all trace data.

Reply:
Could we add a `#singleStatementQuery` method for most cases?

Reply:
Sorry, I make a mistake. This method works for `single statement`. `InfluxClient#query()` is for `multi-statements`